fedimint-server/src/config/api.rs

```diff
@@ -378,6 +428,103 @@ impl ConfigGenApi {
     fn bad_request<T>(msg: &str) -> ApiResult<T> {
         Err(ApiError::bad_request(msg.to_string()))
     }
+
+    pub async fn restart_federation_setup(&self) -> ApiResult<()> {
+        let leader = {
+            let expected_status = [
+                ServerStatus::SharingConfigGenParams,
+                ServerStatus::ReadyForConfigGen,
+                ServerStatus::ConfigGenFailed,
+                ServerStatus::VerifyingConfigs,
+                ServerStatus::VerifiedConfigs,
+            ];
+            let mut state = self.require_any_status(&expected_status)?;
+
+            let cfg_staging_dir = self.data_dir.join(CONFIG_STAGING_DIR);
+            self.remove_staged_configs(&cfg_staging_dir)?;
+
+            state.status = ServerStatus::SetupRestarted;
+            info!(
+                target: fedimint_logging::LOG_NET_PEER_DKG,
+                "Update config gen status to 'Setup restarted'"
+            );
+            // Create a WSClient for the leader
+            state
+                .local
+                .clone()
+                .and_then(|local| local.leader_api_url.map(WsAdminClient::new))
+        };
+
+        self.update_leader().await?;
+
+        // Followers wait for leader to signal that all peers have restarted setup
+        // The leader will signal this by setting it's status to AwaitingPassword
+        let self_clone = self.clone();
+        let mut sub_group = self.task_group.make_subgroup().await;
+        sub_group
+            .spawn("restart", move |_handle| async move {
+                if let Some(client) = leader {
+                    self_clone.await_leader_restart(client).await?;
+                } else {
+                    self_clone.await_peer_restart().await;
+                }
+                // Progress status to AwaitingPassword
+                {
+                    let mut state = self_clone.state.lock().expect("lock poisoned");
+                    state.status = ServerStatus::AwaitingPassword;
```

Comment:
Shouldn't we completely reset the `ConfigGenState` here? Otherwise, wouldn't a leader still use the same set of peers that had registered previously?